Source/WebCore/style/ElementRuleCollector.cpp

```diff
@@ -578,6 +582,69 @@ bool ElementRuleCollector::containerQueriesMatch(const RuleData& ruleData, const
     return true;
 }
 
+bool ElementRuleCollector::scopeQueriesMatch(const RuleData& ruleData, const MatchRequest& matchRequest)
```

Comment:
Let's leave the matching part (everything in ElementRuleCollector) out from the first patch and just focus on the RuletSet and RuleSetBuilder.